src/frontend/src/optimizer/plan_node/stream_exchange.rs

```diff
@@ -34,17 +34,22 @@ pub struct StreamExchange {
 
 impl StreamExchange {
     pub fn new(input: PlanRef, dist: Distribution) -> Self {
-        // Dispatch executor won't change the append-only behavior of the stream.
+        let columns_monotonicity = if matches!(dist, Distribution::Broadcast) {
+            input.columns_monotonicity().clone()
+        } else {
+            // we lost monotonicity information when shuffling
+            MonotonicityMap::new()
+        };
```

Comment:
Does `Broadcast` requires the input to be singleton?

Reply:
Wait, so I guess actually it's when input distribution is `Singleton`, all exchange (not only Broadcast) don't lose monotonicity (if the shuffle _preserves order_).

Reply:
You're right.